be1-go/docs/README.md

```diff
@@ -125,42 +124,41 @@ websocket connections.
 
 ##### Processing messages in the application layer
 
-The incoming messages received by the `ReadPump` are propagated up the stack to
-the `Hub` which is responsible for processing it and sending, depending on the message's nature, a: 
-- `Result` to the request.
-- `Error`
-- `Broadcast`
-- `GreetServer` back to a server that has not been greeted yet.
-- `GetMessagesById` in response to a heartbeat if it is missing some messages.
+The incoming messages received by the `ReadPump` are propagated up the stack to the `Hub`. The `Hub`, on receiving a message,
+processes it by invoking the `HandleIncomingMessage` method from the package `handler` and in case of `Error`, while processing the message, it will log it.
+In parallel, the `Hub` will send a `Heartbeat` every 30 seconds to all the connected servers.
 
-A hub, on receiving a message, processes it by invoking the
-`handleIncomingMessage` method where its handled depending on which `Socket` the
-message originates from.
-
-The flowchart below describes the flow of data and how messages are processed.
+The flowchart below describes all the possible way for a message inside the handlers from package `handler`.
 
 <div align="center">
-  <img src="images/flowchart/flowchart.png" alt="Flowchart"/>
+  <img src="images/handler/handler.png" alt="Flowchart"/>
 </div>
 
 <p align="center"><i>
-  Flowchart last updated at the end of Spring 2023
+  Flowchart last updated at 11.05.2024 and everything in red is still missing in the refactoring
 </i></p>
 
-The hubs themselves contain multiple `Channels` with the `Root` channel being
-the default one, where messages for creation of new LAOs may be published for
-instance. Another example of a channel would be one for an `Election` which
-would be a sub-channel within the LAO channel.
+We use `Socket.SendError` to send an `Error` back to the client. We use this function only in two places, inside `HandleIncomingMessage` 
+in case the format of message is wrong or inside `handleQuery` because we should never answer an error to an answer to avoid loops.
```

Comment:
```suggestion
in case the format of message is wrong, or inside `handleQuery` because we should never answer an error to an answer to avoid loops.
```
Commas help readability :-) 

Also, this documentation reads a lot like describing what the code does - this is something that's easy to gather by reading the code itself. More interesting questions that the documentation should answer is: 
- how : how are responsibilities split across modules ? how are the modules interacting ? etc.
- where, related to the how : where should I change the code to enact a change in feature X or system property Y ? 
- why : why do we do things a certain way ? why are we making a certain compromise ? 

In general, I'd point you to [Code Complete 2nd edition](https://github.com/media-lib/prog_lib/blob/master/general/Steve%20McConnell%20-%20Code%20Complete%20(2nd%20edition).pdf), chapter 32 about documentation :)

Don't get me wrong - the documentation you're replacing was far from perfect, I'm just trying to give you pointers to improve it along the way :-)

Reply:
thx for the book, the docs inside this PR was more in the optic to help you and Mariem to have quickly a global image of the new backend with some diagrams and we already planned to refine the doc before the final presentation.